src/games/services/game-event-handler.service.ts

```diff
@@ -25,6 +32,37 @@ export class GameEventHandlerService implements OnModuleDestroy {
     @Inject('GAME_MODEL_MUTEX') private mutex: Mutex,
   ) {}
 
+  onModuleInit() {
+    this.events.matchStarted.subscribe(
+      async ({ gameId }) => await this.onMatchStarted(gameId),
+    );
+    this.events.matchEnded.subscribe(
+      async ({ gameId }) => await this.onMatchEnded(gameId),
+    );
+    this.events.playerConnected.subscribe(
+      async ({ gameId, steamId }) =>
+        await this.onPlayerJoining(gameId, steamId),
+    );
+    this.events.playerJoinedTeam.subscribe(
+      async ({ gameId, steamId }) =>
+        await this.onPlayerConnected(gameId, steamId),
+    );
+    this.events.playerDisconnected.subscribe(
+      async ({ gameId, steamId }) =>
+        await this.onPlayerDisconnected(gameId, steamId),
+    );
+    this.events.scoreReported.subscribe(
+      async ({ gameId, teamName, score }) =>
+        await this.onScoreReported(gameId, teamName, score),
+    );
+    this.events.logsUploaded.subscribe(
+      async ({ gameId, logsUrl }) => await this.onLogsUploaded(gameId, logsUrl),
+    );
+    this.events.demoUploaded.subscribe(
+      async ({ gameId, demoUrl }) => await this.onDemoUploaded(gameId, demoUrl),
```

Comment:
nit for me in the future, would be nice to have this moved to a decorator.

Reply:
I'm 99% certain with legacy babel decorators I can make one that can bind the class's method to the injected events and then hope that works.

Reply:
And how would that work? Would it be properly typed?

Reply:
Not entirely typed because I can't enforce the method's signature.... but nothing's out of the water yet.

Reply:
Ok, can I merge this as soon as tests are fixed or do you want me to wait for your changes?

Reply:
merge it. I'll think over types (With the millions of other things I Promise ™️ to do)

Reply:
more of a thing that could help reduce the complexity in the future.

Reply:
My API proposal:
```ts
@OnEvent(Events.logsUploaded)
async onLogsUploaded(gameId, logsUrl) { }

@OnEvent(Events.demoUploaded.pipe(
  filter(({ gameId }) => gameId !== undefined),
))
async onDemoUploaded(gameId, demoUrl) { }
```
```

Reply:
Was also thinking about that. That was my exact proposal.